packages/frontend/src/lib/ExamplesCard.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+import Card from './Card.svelte';
+import ExampleCard from './ExampleCard.svelte';
+import type { Example, Category } from '../../../shared/src/models/examples';
+import { onMount } from 'svelte';
+import { bootcClient, rpcBrowser } from '../api/client';
+import type { ImageInfo } from '@podman-desktop/api';
+import { Messages } from '/@shared/src/messages/Messages';
+
+export let category: Category;
+export let examples: Example[];
+let bootcAvailableImages: ImageInfo[] = [];
+
+onMount(async () => {
+  bootcAvailableImages = await bootcClient.listBootcImages();
+  updateExamplesWithPulledImages();
+
+  // Update the available bootc images if we receive a msg image pull update from the UI
+  return rpcBrowser.subscribe(Messages.MSG_IMAGE_PULL_UPDATE, async msg => {
+    bootcAvailableImages = await bootcClient.listBootcImages();
+
+    if (msg.image) {
+      const example = examples.find(example => example.image === msg.image);
+      if (example) {
+        example.state = 'pulled';
+        console.log(examples);
+        examples = [...examples];
+      }
+    }
+  });
+});
+
+// Function to update examples based on available images
+function updateExamplesWithPulledImages() {
+  if (bootcAvailableImages) {
+    bootcAvailableImages.forEach(image => {
+      // Only do it if there is a RepoTags
+      if (image.RepoTags) {
+        const [imageRepo, imageTag] = image.RepoTags[0].split(':');
+        // Find by image name and tag if it's in the list of examples
+        const example = examples.find(example => example.image === imageRepo && example.tag === imageTag);
+        if (example) {
+          example.state = 'pulled';
+          examples = [...examples];
+        }
+      }
+    });
+  }
+}
+
+// Reactive statement to call the function each time bootcAvailableImages updates
+$: if (bootcAvailableImages) {
```

Comment:
might be a time to use $effect / svelte 5 runes

Reply:
Can't as we use `$$props.class` in the Card (copied from AI lab and we try to keep consistent with them too) which we can then not use since it puts the svelte file in "runes mode"

Reply:
NVM, we don't actually use classes imported into it, so it's okay to switch to svelte 5 runes, updating now.

Reply:
I don't see any reason why it would fail